strawberry_django/mutations/fields.py

```diff
@@ -1,21 +1,159 @@
-        mutation_fields[f'create_{object_name}s'] = resolvers.create_batch(model, output_type, input_type)
-        mutation_fields[f'update_{object_name}s'] = resolvers.update(model, output_type, input_type)
-        mutation_fields[f'delete_{object_name}s'] = resolvers.delete(model, output_type, input_type)
-    return strawberry.type(type('Mutation', (), mutation_fields))
-
-mutations.create = resolvers.create
-mutations.create_batch = resolvers.create_batch
-mutations.update = resolvers.update
-mutations.delete = resolvers.delete 
+from strawberry.arguments import StrawberryArgument, UNSET, convert_arguments, is_unset
+from typing import List
+from asgiref.sync import sync_to_async
+from ..resolvers import django_resolver
+from .. import utils, types
+from ..fields.field import StrawberryDjangoFieldBase, StrawberryDjangoFieldFilters, StrawberryField
+from ..fields.types import OneToOneInput, OneToManyInput, ManyToOneInput, ManyToManyInput
+
+from django.db import transaction
+
+class DjangoMutationBase:
+    def __init__(self, input_type, **kwargs):
+        self.input_type = input_type
+        self.many = False
+        super().__init__(graphql_name=None, python_name=None, type_=None, **kwargs)
+
+    def post_init(self):
+        type_ = self.type or self.child.type
+        if self.input_type:
+            assert self.django_model == utils.get_django_model(self.input_type), (
+                'Input and output types should be from the same Django model')
+
+    @property
+    def arguments(self):
+        arguments = []
+        if self.input_type:
+            arguments.append(get_argument('data', self.input_type, self.many))
+        return arguments + super().arguments
+
+    @django_resolver
+    def get_result(self, source, info, kwargs):
+        kwargs = convert_arguments(kwargs, self.arguments)
+        return self.resolver(info=info, source=source, **kwargs)
+
+    def get_wrapped_resolver(self):
+        self.post_init()
+        return super().get_wrapped_resolver()
+
+
+class DjangoCreateMutation(
+        DjangoMutationBase,
+        StrawberryDjangoFieldBase,
+        StrawberryField):
+
+    def post_init(self):
+        if self.is_list:
+            self.many = True
+        super().post_init()
+
+    def create(self, data):
+        input_data = get_input_data(self.input_type, data)
+        instance = self.django_model.objects.create(**input_data)
+        update_m2m([instance], data)
+        return instance
+
+    @transaction.atomic
+    def resolver(self, info, source, data):
+        if self.many:
+            return [self.create(d) for d in data]
+        else:
+            return self.create(data)
+
+
+class DjangoUpdateMutation(
+        DjangoMutationBase,
+        StrawberryDjangoFieldFilters,
+        StrawberryDjangoFieldBase,
+        StrawberryField):
+
+    @transaction.atomic
+    def resolver(self, info, source, data, **kwargs):
+        print('UPDATE', data)
```

Comment:
You forgot a print here :laughing:

Reply:
thanks. this definitely should not be there :)